services/net/syndication/SyndicationAction.cs

```diff
@@ -90,8 +90,8 @@ private async Task ImportFeedAsync(IIngestServiceActionManager manager, Syndicat
                 var reference = await this.FindContentReferenceAsync(manager.Ingest.Source?.Code, item.Id);
                 if (reference == null)
                 {
-                    reference = await AddContentReferenceAsync(manager.Ingest, item);
                     await FetchContent(manager.Ingest, item, link);
+                    reference = await AddContentReferenceAsync(manager.Ingest, item);
```

Comment:
Can't do this as it will cause issues when running services in parallel.  The ContentReference is used as a locking record.

Reply:
Updated, thanks